patches/api/0481-Improve-Riptide-API.patch

```diff
@@ -0,0 +1,131 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: PlayerSchark <smidl115@gmail.com>
+Date: Sun, 26 May 2024 02:56:12 +0200
+Subject: [PATCH] Improve Riptide API
+
+
+diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
+index b777e530122549455dcce6fac8d4a151c1c0af42..dd83153ed7a616aa1388945e48abefafafe0f998 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -801,6 +801,28 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      * @return True if this entity is currently riptiding.
+      */
+     public boolean isRiptiding();
++    // Paper start - Improve Riptide API
++    /**
++     * Starts/Sets the entities riptiding effect for
++     * the specified amount of ticks.<br>
++     * <b>Note: </b> This doesn't currently affect
++     * all entities that extend {@link LivingEntity}.
++     * Some examples include:
++     * <ul>
++     *     <li>{@link EnderDragon}</li>
++     *     <li>{@link ArmorStand}</li>
++     *     <li>and maybe more</li>
++     * </ul>
++     *
++     * @param ticks amount of ticks to be riptiding
++     */
++    public void setRiptiding(int ticks);
++
++    /**
++     * Stops any riptiding currently happening.
++     */
++    public void stopRiptiding();
++    // Paper end - Improve Riptide API
+ 
+     /**
+      * Returns whether this entity is slumbering.
+diff --git a/src/main/java/org/bukkit/event/player/PlayerRiptideEvent.java b/src/main/java/org/bukkit/event/player/PlayerRiptideEvent.java
+index b16fd4433d52951cdd36922678cd4deafc91d789..877b877f8ae74de2b39d5e39c91cb0e9dd3a67d9 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerRiptideEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerRiptideEvent.java
+@@ -1,10 +1,12 @@
+ package org.bukkit.event.player;
+ 
++import net.kyori.adventure.sound.Sound;
```

Comment:
This is not a class created by this patch, please use FQN imports like mentioned in https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md